lib/book.rb

```diff
@@ -0,0 +1,21 @@
+class Book
+  attr_reader :title
+
+  def initialize(book_hash)
+    @title = book_hash[:title]
+    @author = {
+      first_name: book_hash[:author_first_name],  
+      last_name: book_hash[:author_last_name]
+    }
+    @publication_date = book_hash[:publication_date]
+  end
+
+  def author
+    "#{@author[:first_name]} #{@author[:last_name]}"
+  end
+
+  def publication_year
+    @publication_date[-4..-1]
```

Comment:
same thing here, if we don't ever have need of the original date any longer, you could store it as `@publication_year = book_hash[:publication_date][-4..-1]` in the initialize method and make an attr_reader for it. Sometimes it's necessary to "calculate" this kind of thing every time, but sometimes we can pre-calculate things one time